### PR TITLE
feature/se-11-upload-files-over-2gb

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|-----:|-----:|-------:|
| ETA                  |  12.8  |    0 |     0 |      0 |        0 |
| Developing      |  2+8+8+3+1  |    0 |    0 |      0 |         0 |
| Review             |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -   |   -     |        0 |
## Problem description
The admin should be able to upload bitstream bigger than maximum upload size.

- [x] Define a new local metadata type
- [x] Create a test for the new metadata type
- [x] Return upload max file size exceeded response
- [x] Upload patch endpoint
- [x] Create a test to check if is possible to upload file bigger than maximum upload size

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java [hunk 1]
```diff
@@ -84,6 +85,12 @@ protected void handleWrongRequestException(HttpServletRequest request, HttpServl
         sendErrorResponse(request, response, ex, "Request is invalid or incorrect", HttpServletResponse.SC_BAD_REQUEST);
     }
 
+    @ExceptionHandler({MaxUploadSizeExceededException.class})
+    protected void handleMaxFileSizeExceeded(HttpServletRequest request, HttpServletResponse response,
+                                               Exception ex) throws IOException {
+        sendErrorResponse(request, response, ex, "Maximum upload size exceeded", HttpServletResponse.SC_BAD_REQUEST);
```
Comment (vidiecan) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java:91]: https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#SC_REQUEST_ENTITY_TOO_LARGE ?

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/WorkspaceItemRestRepository.java [hunk 2]
```diff
@@ -216,6 +221,12 @@ public void patch(Context context, HttpServletRequest request, String apiCategor
             }
         }
         wis.update(context, source);
+
+        try {
+            uploadFileFromURL(context, request, id);
+        } catch (IOException | URISyntaxException e) {
+            e.printStackTrace();
```
Comment (vidiecan) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/WorkspaceItemRestRepository.java:228]: really like this?

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java [hunk 3]
```diff
@@ -7191,4 +7194,70 @@ public void verifyCorrectInheritanceOfAccessConditionTest() throws Exception {
...
+                    .andExpect(status().isOk())
+                    .andExpect(jsonPath("$.sections.upload.files[0]"
+                                    + ".metadata['dc.title'][0].value",
+                            is(FILE_NAME)))
+                    .andExpect(jsonPath("$.sections.upload.files[0].sizeBytes",
+                            is(SIZE_IN_BYTES.intValue())));
+        } catch (Exception e) {
+            if (!ObjectUtils.isEmpty(file) && file.exists()) {
+                try {
+                    file.delete();
```
Comment (vidiecan) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java:7254]: do you delete the file on success?

File: dspace/config/dspace.cfg [hunk 4]
```diff
@@ -885,6 +885,12 @@ metadata.hide.dc.description.provenance = true
 # Defaults to true; If set to 'false', submitter has option to skip upload
 #webui.submit.upload.required = true
 
+# Maximum size of a single uploaded file
```
Comment (vidiecan) [inline dspace/config/dspace.cfg:888]: indicate we changed this setting by a magic string
`DQUF`

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java (hunk 1 above)
Comment (milanmajchrak) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java:91]: Changed to: SC_REQUEST_ENTITY_TOO_LARGE, thank you

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java (hunk 3 above)
Comment (milanmajchrak) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java:7254]: Added finally clause

**Pull request merged**